### PR TITLE
fix(graph): set schema id by name/message

### PR DESCRIPTION
**Pull request by cea5**
fix bug :schemaName and schemaMessage not set
image

### CODE REVIEW

**Comment by ngjaying** [comment]
Thank you! Please follow the DCO Details link to pass it.

File: internal/topo/planner/planner_graph.go [hunk 1]
```diff
@@ -498,6 +498,7 @@ func parseSource(nodeName string, gn *api.GraphNode, rule *api.Rule, store kv.Ke
 			return nil, ILLEGAL, "", err
 		}
 		sourceOption.TYPE = gn.NodeType
+		sourceOption.SCHEMAID = gn.Props["schemaName"].(string) + "." + gn.Props["schemaMessage"].(string)
```
Comment (ngjaying) [inline internal/topo/planner/planner_graph.go:501]: Should only override schemaId if both schemaName and schemaMessage are set. Additionally, need to check if the props are string type or they will be a panic here.

**Review by ngjaying — CHANGES_REQUESTED** [review]

**Comment by ngjaying** [comment]
LGTM, thank you! Will merge once all checks passed

File: internal/topo/planner/planner_graph.go [hunk 2]
```diff
@@ -498,6 +498,13 @@ func parseSource(nodeName string, gn *api.GraphNode, rule *api.Rule, store kv.Ke
 			return nil, ILLEGAL, "", err
 		}
 		sourceOption.TYPE = gn.NodeType
+		if sourceOption.SCHEMAID == "" && gn.Props["schemaName"] != nil && gn.Props["schemaMessage"] != nil{
```
Comment (ngjaying) [inline internal/topo/planner/planner_graph.go:501]: add a space after nil :  "nil{" -> "nil {"

File: internal/topo/planner/planner_graph.go [hunk 3]
```diff
@@ -498,6 +498,13 @@ func parseSource(nodeName string, gn *api.GraphNode, rule *api.Rule, store kv.Ke
...
+			schemaName, ok1 := gn.Props["schemaName"].(string)
+			schemaMessage, ok2 := gn.Props["schemaMessage"].(string)
+			if ok1 && ok2 {
+				sourceOption.SCHEMAID = schemaName+"."+schemaMessage
```
Comment (ngjaying) [inline internal/topo/planner/planner_graph.go:505]: Add spaces between + :  schemaName + "." + schemaMessage

**Review by ngjaying — CHANGES_REQUESTED** [review]
go fmt is required in this project.

**Review by ngjaying — CHANGES_REQUESTED** [review]
Is it possible for you to add a test case? We can help if you have difficulties.

**Comment by cea5** [comment]
ok, I will try to add one.

File: internal/topo/planner/planner_graph_test.go [hunk 4]
```diff
@@ -583,6 +583,39 @@ func TestPlannerGraphValidate(t *testing.T) {
 }`,
 			err: "",
 		},
+    {
```
Comment (ngjaying) [inline internal/topo/planner/planner_graph_test.go:586]: The indention format is wrong, please fix.

**Review by ngjaying — CHANGES_REQUESTED** [review]

**Review by ngjaying — APPROVED** [review]

**Comment by ngjaying** [comment]
Merged, thank you!

**Pull request merged**